src/sdk/strategy/pre-strategy.ts

```diff
@@ -172,24 +138,44 @@ export class PreStrategy {
         this.bobSecretKey.toBEBytes(),
         other.bobSecretKey.toBEBytes()
       ) &&
-      conditionSetEquals &&
       this.startDate.toString() === other.startDate.toString() &&
       this.endDate.toString() === other.endDate.toString()
     );
   }
 }
 
 export class DeployedPreStrategy {
-  constructor(
-    public label: string,
-    public cohort: Cohort,
-    public policy: EnactedPolicy,
-    public encrypter: Enrico,
-    public decrypter: PreTDecDecrypter,
-    private bobSecretKey: SecretKey,
-    public conditionSet?: ConditionSet
+  private constructor(
+    public readonly cohort: Cohort,
+    private readonly bobSecretKey: SecretKey,
+    public readonly decrypter: PreTDecDecrypter,
+    public readonly policyKey: PublicKey
   ) {}
 
+  public static create(
+    cohort: Cohort,
+    policy: EnactedPolicy,
+    bobSecretKey: SecretKey
```

Comment:
Remind me why `bobSecretKey`? Is this for PreTDec? Base PRE should not need a bob secret key.

I'm a little hazy in this area.

Reply:
Yes, this is for `PreTDecDecrypter`. I think I can further simplify some APIs here.